[SPARK-28810][DOC][SQL] Document SHOW TABLES in SQL Reference.

shivusondur opened the pull request:
<!--
Thanks for sending a pull request!  Here are some tips for you:
  1. If this is your first time, please read our contributor guidelines: https://spark.apache.org/contributing.html
  2. Ensure you have added or run the appropriate tests for your PR: https://spark.apache.org/developer-tools.html
  3. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP][SPARK-XXXX] Your PR title ...'.
  4. Be sure to keep the PR description updated to reflect all changes.
  5. Please write your PR title to summarize what this PR proposes.
  6. If possible, provide a concise example to reproduce the issue for a faster review.
-->

### What changes were proposed in this pull request?
Added the reference for SHOW TABLES sql command.
<!--
Please clarify what changes you are proposing. The purpose of this section is to outline the changes and how this PR fixes the issue. 
If possible, please consider writing useful notes for better and faster reviews in your PR. See the examples below.
  1. If you refactor some codes with changing classes, showing the class hierarchy will help reviewers.
  2. If you fix some SQL features, you can provide some references of other DBMSes.
  3. If there is design documentation, please add the link.
  4. If there is a discussion in the mailing list, please add the link.
-->


### Why are the changes needed?
To help the customer usage
<!--
Please clarify why the changes are needed. For instance,
  1. If you propose a new API, clarify the use case for a new API.
  2. If you fix a bug, you can clarify why it is a bug.
-->


### Does this PR introduce any user-facing change?
It updates the Sql command reference doc.
<!--
If yes, please clarify the previous behavior and the change this PR proposes - provide the console output, description and/or an example to show the behavior difference if possible.
If no, write 'No'.
-->


### How was this patch tested?
<details>
<summary> Attached the Snap</summary>


![image](https://user-images.githubusercontent.com/7912929/66623173-1eac1b80-ec08-11e9-8357-9f6323e5fc48.png)

![image](https://user-images.githubusercontent.com/7912929/65384657-87f3e980-dd42-11e9-90fa-6650ee68e005.png)

</details>
<!--
If tests were added, say they were added here. Please make sure to add some test cases that check the changes thoroughly including negative and positive cases if possible.
If it was tested in a way different from regular unit tests, please clarify how you tested step by step, ideally copy and paste-able, so that other reviewers can test and check, and descendants can verify in the future.
If tests were not added, please describe why they were not added and/or why it was difficult to add.
-->

shivusondur committed:
Updated the document details

shivusondur commented:
SHOW TABLES
![image](https://user-images.githubusercontent.com/7912929/63565811-2a744d80-c588-11e9-858f-0a16820e6b3b.png)

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

amanomer commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,34 @@ license: |
   limitations under the License.
 ---
+### Description
+Return all tables form the database. It shows databse name and whether a table is temporary.
```
form -> from.

amanomer replied:
databse -> database.

amanomer commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,34 @@ license: |
+
+ **LIKE 'pattern'**
+
+ Returns tables which are all match the pattern. In pattern '*' matches any number of characters.
```
Returns or return?

amanomer commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,34 @@ license: |
+SHOW TABLES FROM userdb
+SHOW TABLES IN userdb
+SHOW TABLES FROM userdb LIKE employeedb*
+{% endhighlight %}
```
Add semicolon after each query.

sandeep-katta commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,34 @@ license: |
+
+### Syntax
+{% highlight sql %}
+SHOW TABLES [{FROM|IN} dbname] [LIKE 'pattern']
```
refer [tables.scala](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala#L743) for syntax

shivusondur committed:
Updated the document according to latest comments

shivusondur commented:
@dilipbiswal @gatorsmile 
Updated the PR according to #25525 and #25523 PR's.
Updated the ScreenShot after creating the document from Jekyll build

Please review and kindly check for merging.

gatorsmile commented:
ok to test

SparkQA commented:
**[Test build #111092 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111092/testReport)** for PR 25561 at commit [`172849a`](https://github.com/apache/spark/commit/172849a2bc3465648baf805216fe7d43d70d107e).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/16176/
Test PASSed.

SparkQA commented:
**[Test build #111092 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111092/testReport)** for PR 25561 at commit [`172849a`](https://github.com/apache/spark/commit/172849a2bc3465648baf805216fe7d43d70d107e).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/111092/
Test PASSed.

dilipbiswal commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,73 @@ license: |
+  </dd>
+  <dt><code><em>LIKE 'regex_pattern'</em></code></dt>
+  <dd>
+     A regex pattern that is used to filter out unwanted functions.
```
@shivusondur Can you please check the code here ? Looks like this is copied from `SHOW FUNCTIONS`.  Please make sure the pattern is processed the same way in `SHOW TABLES`. If so, you can change from "unwanted functions" to "unwanted tables"..

shivusondur replied:
Handled

dilipbiswal commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,73 @@ license: |
+{% highlight sql %}
+-- List all tables in default database
+SHOW TABLES;
++-----------+------------+--------------+--+
```
Nit: two space align for results to match the other prs ?

shivusondur replied:
Handled

dilipbiswal commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,73 @@ license: |
+
+{% endhighlight %}
+
+### Related statements.
```
remove period

shivusondur replied:
Done

dilipbiswal commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,73 @@ license: |
+SHOW TABLES [{FROM|IN} database_name] [LIKE 'regex_pattern']
+{% endhighlight %}
+
+### Parameter
```
Nit: Parameters ?

shivusondur replied:
Done

shivusondur committed:
1. changed to "Parameters"
2. Changed to  "unwanted tables"
3. Added the "two space align in the result"
4. removed the period in "statements"
5. verified the links in the Related statements

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/16197/
Test PASSed.

SparkQA commented:
**[Test build #111117 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111117/testReport)** for PR 25561 at commit [`19f39fc`](https://github.com/apache/spark/commit/19f39fcfad436a7a6016005910e0c55f998a8950).

shivusondur commented:
@dilipbiswal 
i handled all your comments, please have a look.
and again thanks for your review.

SparkQA commented:
**[Test build #111117 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111117/testReport)** for PR 25561 at commit [`19f39fc`](https://github.com/apache/spark/commit/19f39fcfad436a7a6016005910e0c55f998a8950).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/111117/
Test PASSed.

dilipbiswal commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,73 @@ license: |
+  <dt><code><em>LIKE 'regex_pattern'</em></code></dt>
+  <dd>
+     Specifies the regex pattern that is used to filter out unwanted tables.
+       <br> - Only `*` and `|` are allowed as wildcard pattern.
```
@shivusondur Can you please include the example with output using the 'I' as wildcard ? From the code its not clear to me if this is supported.

shivusondur replied:
done

dilipbiswal commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,73 @@ license: |
+{% highlight sql %}
+-- List all tables in default database
+SHOW TABLES;
+  +-----------+------------+--------------+--+
```
Please lets fix the formatting. Did you construct this output by hand ? Can we paste the
actual output here please so we preserve the formatting.

shivusondur replied:
@dilipbiswal 
I just Copy and paste the actual output from the Beeline console
Here is the beeline console snap
![image](https://user-images.githubusercontent.com/7912929/65381823-68df6280-dd16-11e9-873c-c49244e7aee7.png)

In the UI it looks like below, i think it is normal
![image](https://user-images.githubusercontent.com/7912929/65381847-08045a00-dd17-11e9-9d3f-69ec058eaa98.png)

Only in the git file editor, it shows inconsistent.

shivusondur committed:
1. Added the one more example to demonstrate the "|" wild card

SparkQA commented:
**[Test build #111140 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111140/testReport)** for PR 25561 at commit [`55ea2cf`](https://github.com/apache/spark/commit/55ea2cf1c6bf07e9d033a78f762bae42968bfe4c).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/16219/
Test PASSed.

SparkQA commented:
**[Test build #111140 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111140/testReport)** for PR 25561 at commit [`55ea2cf`](https://github.com/apache/spark/commit/55ea2cf1c6bf07e9d033a78f762bae42968bfe4c).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/111140/
Test PASSed.

dilipbiswal commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,82 @@ license: |
+  | default   | sam        | false        |
+  | default   | sam1       | false        |
+  +-----------+------------+--------------+--+
+-- List all tables matching the pattern `sam*|suj`
```
Nit: can we have a line between previous test and this one.

shivusondur replied:
@dilipbiswal 
Done, updated the snap's also.

shivusondur committed:
Added the space between examples

SparkQA commented:
**[Test build #111148 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111148/testReport)** for PR 25561 at commit [`0dd4efb`](https://github.com/apache/spark/commit/0dd4efb71e8a921cc0c57534fc8492803a8b65be).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/16227/
Test PASSed.

SparkQA commented:
**[Test build #111148 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111148/testReport)** for PR 25561 at commit [`0dd4efb`](https://github.com/apache/spark/commit/0dd4efb71e8a921cc0c57534fc8492803a8b65be).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/111148/
Test PASSed.

dilipbiswal commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,83 @@ license: |
+### Description
 
-**This page is under construction**
+List all the `tables` from the database with `database-name` and `is temporary table`.
```
@shivusondur Sorry.. re-reading this one last time :-)
`SHOW TABLES` statement returns all the tables for an optionally specified database. Additionally, the output
of this statement may be filtered via an optional matching pattern. If no database is specified then the tables
are returned from the current database.

shivusondur replied:
DOne

dilipbiswal commented:
LGTM - except a minor comment.

dilipbiswal commented:
cc @gatorsmile

shivusondur committed:
Updated the description

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/16256/
Test PASSed.

dilipbiswal commented:
@shivusondur Thanks for working on it.
LGTM
cc @gatorsmile

shivusondur commented:
@dilipbiswal and @gatorsmile 
Handled the all the comments.

SparkQA commented:
**[Test build #111176 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111176/testReport)** for PR 25561 at commit [`1d08b22`](https://github.com/apache/spark/commit/1d08b225f23e382390e95a07ce09eddb6e57cc53).

SparkQA commented:
**[Test build #111176 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111176/testReport)** for PR 25561 at commit [`1d08b22`](https://github.com/apache/spark/commit/1d08b225f23e382390e95a07ce09eddb6e57cc53).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/111176/
Test PASSed.

srowen commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,86 @@ license: |
+  <dt><code><em>LIKE 'regex_pattern'</em></code></dt>
+  <dd>
+     Specifies the regex pattern that is used to filter out unwanted tables.
+       <br> - Only `*` and `|` are allowed as wildcard pattern.
```
`|` isn't a wildcard pattern, so I wouldn't imply it is. 
Is it really treated as a full regex otherwise? if not, I wouldn't call it a regex unless it is generally called this in other similar references.

dilipbiswal replied:
@srowen Thats a very good point Sean. How about we name the pattern 'identifier_with_wildcards' (and describe as such) as mentioned in `ShowTablesCommand`'s comment ?

srowen replied:
Is it a SQL-like pattern that supports wildcards like `%`? then I'd just call it a `pattern`

dilipbiswal replied:
@srowen I quickly tried in my env. It does not support SQL-like pattern like `%`. I only saw the following work :
- `*` : Matches 0 or more characters.
- `.` : Matches a char
- `|` : works like or expression.

@shivusondur Can you please confirm ?

shivusondur replied:
@dilipbiswal @srowen 
I also tesed, it is same behaviour you mentioned above.
It will use the below method to handle, here they are handling "|" and "*" seperatly, remaining treating as regex
https://github.com/apache/spark/blob/e1ea806b3075d279b5f08a29fe4c1ad6d3c4191a/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala#L88

srowen replied:
OK, your comment is consistent with the docs then. Hm, but isn't it more accurate to say that `*` and `|` _don't_ work like in regexes? `*` matches 0 or more characters, and `|` delimits whole regexes, which is different, but the rest is the same. Maybe a comment more like that. "The pattern is a regex except that `*` matches 0 or more characters by itself, and `|` may only be used at the top level to delimit alternative regexes to match." ?

shivusondur replied:
@srowen @dilipbiswal 
Thank you both for your time in reviewing.

I updated according to your suggestion and screenshot also I updated.
Now it will look like below.
![image](https://user-images.githubusercontent.com/7912929/65989006-847d0280-e4a6-11e9-9b9a-3e9ebcaf92f8.png)

shivusondur committed:
Updated the description

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/16651/
Test PASSed.

SparkQA commented:
**[Test build #111655 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111655/testReport)** for PR 25561 at commit [`847fb8e`](https://github.com/apache/spark/commit/847fb8ee3167a89c9796c435b38b46140080036a).

SparkQA commented:
**[Test build #111655 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111655/testReport)** for PR 25561 at commit [`847fb8e`](https://github.com/apache/spark/commit/847fb8ee3167a89c9796c435b38b46140080036a).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/111655/
Test PASSed.

srowen commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,86 @@ license: |
+### Description
 
-**This page is under construction**
+`SHOW TABLES` statement returns all the tables for an optionally specified `database`.
```
Nit: add "The" at the start.
via -> by

shivusondur replied:
Done

shivusondur committed:
handled the comments

SparkQA commented:
**[Test build #111750 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111750/testReport)** for PR 25561 at commit [`05a2e60`](https://github.com/apache/spark/commit/05a2e600c48f934864da0224694f5572cbb62b13).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/16737/
Test PASSed.

SparkQA commented:
**[Test build #111750 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111750/testReport)** for PR 25561 at commit [`05a2e60`](https://github.com/apache/spark/commit/05a2e600c48f934864da0224694f5572cbb62b13).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/111750/
Test PASSed.

srowen commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,86 @@ license: |
+  <dt><code><em>LIKE 'regex_pattern'</em></code></dt>
+  <dd>
+     Specifies the regex pattern that is used to filter out unwanted tables.
+       <br> - The pattern is a regex except that `*` and `|`characters
```
"... except that `*` and `|` work differently"

srowen commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,86 @@ license: |
+### Description
 
-**This page is under construction**
+The `SHOW TABLES` statement returns all the tables for an optionally specified `database`.
```
Nit: no need to back-tick database here

shivusondur replied:
@srowen 
Done

dilipbiswal commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,86 @@ license: |
+  <dt><code><em>LIKE 'regex_pattern'</em></code></dt>
+  <dd>
+     Specifies the regex pattern that is used to filter out unwanted tables.
+       <br> - The pattern is a regex except that `*` and `|`characters
```
@shivusondur drop the "that" ? 
@srowen Can you help check the above sentence please ?

shivusondur replied:
@dilipbiswal 
done

shivusondur committed:
handled the comments

SparkQA commented:
**[Test build #111830 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111830/testReport)** for PR 25561 at commit [`fed9431`](https://github.com/apache/spark/commit/fed9431529a9b1eeb9b709d56902f24ef62fc0dc).

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/16809/
Test PASSed.

SparkQA commented:
**[Test build #111830 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/111830/testReport)** for PR 25561 at commit [`fed9431`](https://github.com/apache/spark/commit/fed9431529a9b1eeb9b709d56902f24ef62fc0dc).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/111830/
Test PASSed.

dilipbiswal commented:
LGTM
cc @srowen @gatorsmile

srowen commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,86 @@ license: |
+  <dt><code><em>LIKE 'regex_pattern'</em></code></dt>
+  <dd>
+     Specifies the regex pattern that is used to filter out unwanted tables.
+       <br> - The pattern is a regex except `*` and `|`characters
```
Nit: can you use an HTML unordered list? `<ul><li>...` etc? Or just make this a paragraph, not bullet points.

I still think this is kind of unclear, per the last comment. "except `*` ..." should be clarified. "except that `*` and `|` have a different meaning."

shivusondur replied:
done.
used the html ul and li  tags  
and added the description like below
 `Except `*` and `|` characters remaining characters will follow the regular expression convention.`

shivusondur committed:
handled the comments

srowen commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -38,10 +38,13 @@ SHOW TABLES [{FROM|IN} database_name] [LIKE 'regex_pattern']
-       <br> - The leading and trailing blanks are trimmed in the input pattern before processing.  
+     Specifies the regular expression pattern that is used to filter out unwanted tables. 
+     <ul> 
+          <li> Except `*` and `|` characters remaining characters will follow the regular expression convention.</li>
```
"Except for `*` and `|`, the pattern works like a regex"

srowen commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -38,10 +38,13 @@ SHOW TABLES [{FROM|IN} database_name] [LIKE 'regex_pattern']
+     Specifies the regular expression pattern that is used to filter out unwanted tables. 
+     <ul> 
+          <li> Except `*` and `|` characters remaining characters will follow the regular expression convention.</li>
+          <li> `*` matches 0 or more characters and `|` used to provide more than one regex with OR condition. </li>
```
"`*` alone matches 0 or more characters, and `|` is used to separate multiple different regexes, any of which can match"

shivusondur replied:
comments handled
Thanks for reviewing

shivusondur committed:
handled the comments

shivusondur committed:
handled the comments

shivusondur committed:
handled the comments

shivusondur commented on docs/sql-ref-syntax-aux-show-tables.md:
```diff
@@ -18,5 +18,90 @@ license: |
+  <dd>
+     Specifies the regular expression pattern that is used to filter out unwanted tables. 
+     <ul> 
+          <li> Except for * and | character, the pattern works like a regex.</li>
```
Done. The same correction I also interested. thank you

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder-K8s/16945/
Test PASSed.

srowen commented:
Merged to master
